### PR TITLE
Update checkout tests to use API and wait on UI

### PR DESCRIPTION
**Pull request by zhongruige**
### Changes proposed in this Pull Request:

This PR updates the component function for e2e tests to use the API to create shipping zones. As this is used across multiple tests, the changes are very surgical in nature. Additionally, this PR updates the calculate shipping tests to wait more on the UI, including when reloading the page.

### How to test the changes in this Pull Request:

1. Verify the CI passes
2. Run the calculate shipping test locally and verify it passes:

```
npx wc-e2e test:e2e specs/front-end/cart-calculate-shipping.test.js
```

### CODE REVIEW

**Comment by rrennick** [comment]
@zhongruige The shipping test fails for me locally. The general settings are limited to selling to US customers. I commented out the `afterAll()`. Here's a screenshot of the France shipping method:

Screen Shot 2021-08-16 at 3 55 15 PM

File: tests/e2e/core-tests/specs/shopper/front-end-cart-calculate-shipping.test.js [hunk 1]
```diff
@@ -43,29 +39,19 @@ const runCartCalculateShippingTest = () => {
 			await createSimpleProduct(firstProductName);
 			await createSimpleProduct(secondProductName, secondProductPrice);
 
-			await merchant.login();
-			await merchant.openNewShipping();
-
 			// Add a new shipping zone Germany with Free shipping
```
Comment (rrennick) [inline tests/e2e/core-tests/specs/shopper/front-end-cart-calculate-shipping.test.js:44]: I got the tests to pass by adding a reset before adding the shipping zones

```
			await withRestApi.resetSettingsGroupToDefault( 'general' );
```

```
Test Suites: 1 skipped, 38 passed, 38 of 39 total
Tests:       2 skipped, 164 passed, 166 total
Snapshots:   0 total
Time:        752.343s
Ran all test suites.
Jest exit code: 0
```

File: tests/e2e/utils/src/flows/with-rest-api.js [hunk 2]
```diff
@@ -93,6 +93,80 @@ export const withRestApi = {
...
+		additionalZoneMethods = [] ) => {
+
+	   const path = 'wc/v3/shipping/zones';
+
+	   const response = await client.post( path, { name: zoneName } );
+	   expect(response.statusCode).toEqual(201);
+	   let zoneId = response.data.id;
+
+	   // Select shipping zone location
+	   let zoneType = zoneLocation.split(/:(.)/)[0];
```
Comment (rrennick) [inline tests/e2e/utils/src/flows/with-rest-api.js]: These two lines can be replaced with

```
	   let [ zoneType, zoneCode ] = zoneLocation.split(/:(.+)/);
```

File: tests/e2e/utils/src/flows/with-rest-api.js [hunk 3]
```diff
@@ -93,6 +93,80 @@ export const withRestApi = {
...
+		   }
+	   ];
+
+	   // Fill shipping zone postcode if provided
+	   if ( zipCode ) {
+		   zoneLocationPayload.push( {
+			   code: zipCode,
+			   type: "postcode",
+		   } );
+	   };
```
Comment (rrennick) [inline tests/e2e/utils/src/flows/with-rest-api.js]: Extraneous comma.

File: tests/e2e/core-tests/specs/shopper/front-end-cart-calculate-shipping.test.js [hunk 4]
```diff
@@ -84,13 +71,14 @@ const runCartCalculateShippingTest = () => {
 		});
 
 		it('allows customer to calculate Flat rate and Local pickup if in France', async () => {
-			await page.reload();
+			await page.reload( { waitUntil: ['networkidle0', 'domcontentloaded'] });
```
Comment (rrennick) [inline tests/e2e/core-tests/specs/shopper/front-end-cart-calculate-shipping.test.js]: Inconsistent spacing.

File: tests/e2e/utils/src/components.js [hunk 5]
```diff
@@ -419,7 +419,7 @@ const createCoupon = async ( couponAmount = '5', discountType = 'Fixed cart disc
  * @param zipCode Shipping zone zip code. Defaults to empty one space.
  * @param zoneMethod Shipping method type. Defaults to flat_rate (use also: free_shipping or local_pickup)
  */
-const addShippingZoneAndMethod = async ( zoneName, zoneLocation = 'country:US', zipCode = ' ', zoneMethod = 'flat_rate' ) => {
+ const addShippingZoneAndMethod = async ( zoneName, zoneLocation = 'country:US', zipCode = ' ', zoneMethod = 'flat_rate' ) => {
```
Comment (rrennick) [inline tests/e2e/utils/src/components.js]: Stray leading space.

**Review by rrennick — CHANGES_REQUESTED** [review]
@zhongruige Nice work on this one. I've added some minor change suggestions.

File: tests/e2e/utils/src/components.js (hunk 5 above)
Comment (zhongruige) [reply tests/e2e/utils/src/components.js]: Good catch, removed.

File: tests/e2e/core-tests/specs/shopper/front-end-cart-calculate-shipping.test.js (hunk 4 above)
Comment (zhongruige) [reply tests/e2e/core-tests/specs/shopper/front-end-cart-calculate-shipping.test.js]: Thanks, fixed!

File: tests/e2e/utils/src/flows/with-rest-api.js (hunk 3 above)
Comment (zhongruige) [reply tests/e2e/utils/src/flows/with-rest-api.js]: Removed 👍

File: tests/e2e/utils/src/flows/with-rest-api.js (hunk 2 above)
Comment (zhongruige) [reply tests/e2e/utils/src/flows/with-rest-api.js]: Thanks for the feedback! Changed.

File: tests/e2e/core-tests/specs/shopper/front-end-cart-calculate-shipping.test.js (hunk 1 above)
Comment (zhongruige) [reply tests/e2e/core-tests/specs/shopper/front-end-cart-calculate-shipping.test.js:44]: Oh good idea, I just added that in and they ran successfully for me, too:

```
Test Suites: 1 skipped, 38 passed, 38 of 39 total
Tests:       2 skipped, 164 passed, 166 total
Snapshots:   0 total
Time:        791.236s
Ran all test suites.
Jest exit code: 0
```

Pushed up changes including that line

File: tests/e2e/utils/src/flows/with-rest-api.js (hunk 3 above)
Comment (rrennick) [reply tests/e2e/utils/src/flows/with-rest-api.js]: @zhongruige Sorry, I meant to say extraneous semi-colon on the end of the `if` block. The commas in the object declarations were okay.

**Review by rrennick — DISMISSED** [review]
@zhongruige This worked great locally for me as well. This is good for merge other than fixing the comma/semi-colon error due to my earlier comment.

```
Test Suites: 1 skipped, 38 passed, 38 of 39 total
Tests:       2 skipped, 164 passed, 166 total
Snapshots:   0 total
Time:        805.347s
Ran all test suites.
Jest exit code: 0
```

File: tests/e2e/utils/src/flows/with-rest-api.js (hunk 3 above)
Comment (zhongruige) [reply tests/e2e/utils/src/flows/with-rest-api.js]: No worries! Updated :D

**Review by rrennick — APPROVED** [review]
Works great.

**Pull request merged**